### PR TITLE
Feat: add tokenomics documentation #NTRN-333 #NTRN-334 #NTRN-335

### PR DESCRIPTION
**Pull request by foxpy**

### CODE REVIEW

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
There are no links to the contracts in the repo. Would be nice if:
* each message/query has links to the corresponding handler in the contracts code;
* overview of each contract has links to the corresponding logic in the code (especially treasury and distribution contract overviews);
* some links to the pausing logic in the code.

File: docs/neutron/tokenomics/treasury/messages.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Messages
```
[the remark this answers is not in the document]
Comment (NeverHappened) [reply docs/neutron/tokenomics/treasury/messages.md:1]: Yeah, and to be the same everywhere lets use following structure:
# Messages
## Instantiate
## Messages (?)
### Message1
### Message2

[the remark this answers is not in the document]
Comment (foxpy) [reply docs/neutron/tokenomics/treasury/messages.md:1]: I would replace "Messages (?)" with ExecuteMsg

File: docs/neutron/tokenomics/reserve/messages.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+### TransferOwnership
+
+Transfer the contract's ownership to another account. Can be executed by `main_dao_address` only.
+
+### Payout
+
+Send specified `amount` of funds to the `recipient` address. Can be executed by `main_dao_address` only.
+
```
Comment (foxpy) [inline docs/neutron/tokenomics/reserve/messages.md:52]: And now we have both code comments in enum declaration and standalone articles per each message. Please pick one, having both at the same time doesn't look great.

Comment (NeverHappened) [reply docs/neutron/tokenomics/reserve/messages.md:52]: I don't know what to pick. I like code ones more, but you cannot link them from other places

Comment (pr0n00gler) [reply docs/neutron/tokenomics/reserve/messages.md:52]: Why can't we have both? The difference is standalone articles should be more informative

File: docs/neutron/tokenomics/treasury/overview.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+To calculate how many coins we distribute (release) we use following expression: `burnt_tokens * a_multiplier`. The `a_multiplier` is a linear function: fn(y) = x / `<configurable denominator>`. Configurable denominator should be set based on coins burning rate and preferred total duration of vesting. Take into account that while initially one burnt tokens equals multiple NTRN tokens made liquid, the flow of new tokens into the Treasury progressively slows down until the tokens supply is exhausted and the tokenomy becomes deflationary.
+
+Treasury contract can be configured only by DAO. The contract has the `min_period` parameter, and the `distribute` method cannot be called more than once every `min_period`. The `distribute` call is permissionless and can be called by anybody.
+
+## Deployment
+
+This is one of contracts that are prebaked and initialized into Neutron genesis.
+
+Initialization message contains Main DAO and Security DAO address.
```
[the remark this answers is not in the document]
Comment (albertandrejev) [reply docs/neutron/tokenomics/treasury/overview.md]: what docs?

**Comment by albertandrejev** [comment]
> There are no links to the contracts in the repo. Would be nice if:

> each message/query has links to the corresponding handler in the contracts code;
> overview of each contract has links to the corresponding logic in the code (especially treasury and distribution contract overviews);
> some links to the pausing logic in the code.

It is hard to have links to the code because these links becomes outdated very quickly. Maybe we can add links to the contracts itself or to the file containing required code (but that doesn't fix the problem completely.)

File: docs/neutron/tokenomics/treasury/messages.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    /// Address of the security DAO contract
+    security_dao_address: Option<String>,
+
+    /// Denominator used in the vesting release function
+    vesting_denominator: Option<u128>,
+```
+
+### Pause
+
+Pause contract for `duration` amount of blocks. Permissioned can be executed only by Main DAO or the Security DAO. If contract is in paused state it disables `exectue` method processing for any message except `Pause` and `Unpause`.
```
Comment (pr0n00gler) [inline docs/neutron/tokenomics/treasury/messages.md:96]:
```suggestion
Pause contract for `duration` amount of blocks. Permissioned can be executed only by Main DAO or the Security DAO. If contract is in paused state it disables `execute` method processing for any message except `Pause` and `Unpause`.
```

**Pull request merged**